### DISCUSSION TITLE
Custom command: copy filename

### DISCUSSION THREAD

**Discussion by woezel1980** [hsoft/dupeguru#1071] [Q&A]
Could someone help me with a custom command to copy filenames from the references to the duplicates?
It should be somewhat error restistant to create an alternative name when one already exists (e.g. by adding "(1)" or "(2)" or so).
And it should only copy the filename without the extension.

Thanks in advance!

**Comment by Ikkoru** [comment]
I'm not a scripter, but I managed to find and edit some options for Custom Command renaming:
- Rename the **duplicate** to be the same as the **reference** (have **only** the duplicates selected):
`powershell Rename-Item -LiteralPath '%d' -NewName ([System.IO.Path]::GetFileName('%r'))`

- Rename the **reference** to be the same as the **duplicate** (have **only** the duplicates selected):
`powershell Rename-Item -LiteralPath '%r' -NewName ([System.IO.Path]::GetFileName('%d'))`

- Append the **reference** to the **duplicate** (have **only** the duplicates selected):
`powershell -Command "Rename-Item -LiteralPath '%r' -NewName (([System.IO.Path]::GetFileNameWithoutExtension('%r')).ToString() + ' ' + ([System.IO.Path]::GetFileNameWithoutExtension('%d')).ToString() + ([System.IO.Path]::GetExtension('%r')).ToString())"`

- Append the **duplicate** to the **reference** (have **only** the duplicates selected):
`powershell -Command "Rename-Item -LiteralPath '%d' -NewName (([System.IO.Path]::GetFileNameWithoutExtension('%d')).ToString() + ' ' + ([System.IO.Path]::GetFileNameWithoutExtension('%r')).ToString() + ([System.IO.Path]::GetExtension('%d')).ToString())"`

It's not perfectly to your specifications, but I suck so I hope you can forgive me.
